### PR TITLE
GDPR support

### PR DESCRIPTION
**Pull request by anweshan**
Adds support for 
- detailedResponse (allows you to return response headers) [WIP]
- headers for each method (you can pass in custom headers in each method) (complete)

Note: Assistant tests are commented out at the minute as I am working on detailedResponse

### CODE REVIEW

File: assistant/v1.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
+ *      http://www.apache.org/licenses/LICENS[params.headers] -2.0
```
Comment (germanattanasio) [inline assistant/v1.ts]: We probably want to remove this

File: assistant/v1.ts [hunk 2]
```diff
@@ -107,11 +109,11 @@ class AssistantV1 extends BaseService {
         path,
       },
       defaultOptions: extend(true, {}, this._options, {
-        headers: {
+        headers: extend(true,{
```
Comment (germanattanasio) [inline assistant/v1.ts]: leave space after `true,`

File: test/unit/test.assistant.v1.js [hunk 3]
```diff
@@ -67,6 +67,7 @@ describe('assistant-v1', function() {
   };
 
   const assistant = new watson.AssistantV1(service);
+  const assistant2 = new watson.AssistantV1(extend({}, service, { detailedResponse: true }));
```
Comment (germanattanasio) [inline test/unit/test.assistant.v1.js]: `assistantWithResponseHeaders` it's a better name than `assistant2`

File: test/unit/test.assistant.v1.js [hunk 4]
```diff
@@ -154,6 +155,19 @@ describe('assistant-v1', function() {
       );
       assert.equal(req.method, 'DELETE');
     });
+
+    it('should generate a valid payload with detailed response', function() {
+      const params2 = { text: 'foo', workspace_id: 'foo', headers: { customheader: 'custom' } };
```
Comment (germanattanasio) [inline test/unit/test.assistant.v1.js]: don't use `params2`

File: test/unit/test.assistant.v1.js [hunk 5]
```diff
@@ -154,6 +155,19 @@ describe('assistant-v1', function() {
...
+      const req = assistant2.deleteCounterexample(params2, function(err, res) {
+        console.log(JSON.stringify(res));
```
Comment (germanattanasio) [inline test/unit/test.assistant.v1.js]: Here you probably want to check that the response headers are there.

**Comment by anweshan** [comment]
PR is updated

I added integration test checking that we can get the response headers, but I couldn't explicitly check this in the unit test as of now

File: tone-analyzer/v3-generated.ts [hunk 6]
```diff
@@ -115,8 +114,9 @@ class ToneAnalyzerV3 extends BaseService {
...
-   * @param {string} [params.accept_language] - The desired language of the response. For two-character arguments, regional variants are treated as their parent language; for example, `en-US` is interpreted as `en`.
+   * @param {string} [params.content_language] - The language of the input text for the request: English or French. Regional variants are treated as their parent language; for example, `en-US` is interpreted as `en`. The input content must match the specified language. Do not submit content that contains both languages. You can use different languages for **Content-Language** and **Accept-Language**. * **`2017-09-21`:** Accepts `en` or `fr`. * **`2016-05-19`:** Accepts only `en`.
+   * @param {string} [params.accept_language] - The desired language of the response. For two-character arguments, regional variants are treated as their parent language; for example, `en-US` is interpreted as `en`. You can use different languages for **Content-Language** and **Accept-Language**.
+   * @param {Object} [params.headers] - Custom request headers
```
Comment (germanattanasio) [inline tone-analyzer/v3-generated.ts]: When describing the GDPR support please make sure you communicate that users may override headers that we use as parameters.

If they send {

```js
{ params: {
    accept_language: 'en'
    headers: {
      'Accept-Language': 'batman'
    }
  }
}
```
Batman will take presedence.

```js
      defaultOptions: extend(true, {}, this._options, {
        headers: {
        headers: extend(true, {
          'Accept': 'application/json',
          'Content-Type': 'multipart/form-data',
        }
      })
        }, _params.headers),
      }),
```

Comment (anweshan) [reply tone-analyzer/v3-generated.ts]: Ok, I updated the README.

**Pull request merged**